projects/laji/src/app/+trait-db/shared/trait-search/trait-search-filters/additional-filters.component.ts

```diff
@@ -81,62 +81,84 @@ export type AdditionalFilterValues = Partial<{
 }>;
 
-export const stripDefaultValues = (f: AdditionalFilterValues): Partial<AdditionalFilterValues> => {
-  const newFilters: Partial<AdditionalFilterValues> = {};
-  Object.entries(f).forEach(([k, v]) => {
-    const formKey = propToFormKey[k];
-    if (
-      v !== undefined
-      && v !== filterDefaultValues[formKey]
-      && !(Array.isArray(v) && v.length === 0)
-    ) {
-      newFilters[k as keyof AdditionalFilterValues] = v as any;
-    }
-  });
-  return newFilters;
-};
-
 @Component({
   selector: 'laji-trait-search-additional-filters',
   templateUrl: './additional-filters.component.html',
-  styleUrls: ['./additional-filters.component.scss']
+  styleUrls: ['./additional-filters.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => TraitSearchAdditionalFiltersComponent),
+      multi: true
+    }
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class TraitSearchAdditionalFiltersComponent implements OnInit {
-  @Input() initialValue?: AdditionalFilterValues;
-  @Output() filterChange: Observable<AdditionalFilterValues>;
-
+export class TraitSearchAdditionalFiltersComponent implements ControlValueAccessor, OnDestroy {
   form: FormGroup<Record<keyof typeof additionalFilters, FormControl>>;
   additionalFilters!: AdditionalFilterWithFormKey[];
   additionalFiltersLookup = additionalFilters;
   unselectedAdditionalFilters = new Set<number>();
   selectedAdditionalFilters = new Set<number>();
 
-  constructor(private fb: FormBuilder) {
+  onChange: (value: AdditionalFilterValues | null) => void = () => {};
+  onTouched: () => void = () => {};
+  subscription!: Subscription;
+
+  constructor(private fb: FormBuilder, private cdr: ChangeDetectorRef) {
     this.form = this.fb.group(filterDefaultValues);
-    this.filterChange = this.form.valueChanges.pipe(
-      map(values =>
-        Object.entries(values)
-          .reduce((p, [k, v]) => { p[additionalFilters[k as unknown as keyof typeof additionalFilters].prop] = v; return p; }, {} as AdditionalFilterValues)
-      )
-    );
     this.additionalFilters = additionalFilterArr;
     this.additionalFilters.forEach((item, idx) => {
       this.unselectedAdditionalFilters.add(idx);
     });
+    this.subscription = this.form.valueChanges.pipe(
+      map(values =>
+        Object.entries(values)
+          .reduce((p, [k, v]) => {
+            if (values[k]) {
+              p[additionalFilters[k as unknown as keyof typeof additionalFilters].prop] = v;
+            }
+            return p;
+          }, {} as AdditionalFilterValues)
+      ),
+    ).subscribe(val => this.onChange(val));
   }
 
-  ngOnInit(): void {
-    if (this.initialValue) {
-      const propsMappedToFormKeys: any = {};
-      Object.entries(this.initialValue).forEach(([k, v]) => {
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  writeValue(obj: any): void {
+    const propsMappedToFormKeys: any = {};
```

Comment:
`any` types could be replaced with a stricter type if possible, this could probably be `Record<string, any>`?